### PR TITLE
Add lantern ip tool

### PR DESCRIPTION
**Pull request by AGMETEOR**
Introduce IP tool to lantern-shadowsocks

### CODE REVIEW

File: net/private_net.go [hunk 1]
```diff
@@ -44,7 +46,9 @@ func IsPrivateAddress(ip net.IP) bool {
 			return true
 		}
 	}
-	return false
+
+	ipt, _ := iptool.New()
+	return ipt.IsPrivate(&net.IPAddr{IP: ip})
```
Comment (hwh33) [inline net/private_net.go:51]: This is concise and works, but I suggest we instead add to the existing private ranges defined in this file (in the `init` function).  This way we can try to get the fix merged upstream - which we should definitely do IMO.  We also avoid adding redundant logic this way.

Comment (hwh33) [reply net/private_net.go:51]: To be clear, I'm saying that we should just take the private ranges [defined in `iptool`](https://github.com/getlantern/iptool/blob/c00e863b2696eeb5fe4fb20b33c7918d69bbb9ba/iptool.go#L15-L43) and add them to the ranges defined in the `init` function here.

**Review by hwh33 — COMMENTED** [review]
Sorry @AGMETEOR, this PR completely dropped off my radar!  I'll keep an eye out for future updates and be more responsive going forward.

File: net/private_net.go (hunk 1 above)
Comment (AGMETEOR) [reply net/private_net.go:51]: Sorry @hwh33 this had fallen off my radar. Going to implement this feedback.

**Comment by myleshorton** [comment]
I actually think their assessment is wrong. [RequirePublicIP](https://github.com/getlantern/lantern-shadowsocks/blob/main/net/private_net.go#L55) checks `IsGlobalUnicast`, which does check for 169 addresses.

**Comment by myleshorton** [comment]
Closing since the finding is just incorrect.

**Pull request closed without merging**